File: src/machine/SxxMachine/Continuation.cpp
using namespace std;

#include "Continuation.h"
#include "Operation.h"
#include "Term.h"
#include "Code.h"
#include "Data.h"
#include "Appendable.h"

namespace SxxMachine {

	Continuation::Continuation(std::vector<Term*>& args, Code* c) {
		int i = c->Arity() + 1;
		this->Args = std::vector<Term*>(i);
		while(i-- > 0) {
			this->Args[i] = args[i];
		}
		this->code = c;
	}

	Continuation::Continuation(std::vector<Term*>& args, const int& getArity, Operation* c) {
		int i = getArity + 1;
		this->Args = std::vector<Term*>(i);
		while(i-- > 0) {
			this->Args[i] = args[i];
		}
		this->code = c;
	}

	Term* Continuation::Deref() {
		return this;
	}

	bool Continuation::Unify(Term* that, Prolog* mach) {
		return that->Bind(this, mach);
	}

	void Continuation::formattedOutput(const int& formatFlags, Appendable* buffer) throw(IOException) {
		cout << "general output of continuations not available" << endl;
		buffer->append(this->GetName());
		int i = this->Args.size();
		int current = 0;
		while(i-- > 0) {
			this->formattedArgOutput(this->Args[current++], formatFlags, buffer);
		}

	}

	void Continuation::formattedArgOutput(Term* term, const int& formatFlags, Appendable* buffer) throw(IOException) {
		if(term != nullptr) {
			term->formattedOutput(formatFlags, buffer);
		}

	}

	int Continuation::Arity() {
		return Data::OBJ;
	}
}

File: jsrc/machine/SxxMachine/Continuation.java
package SxxMachine;

import java.io.IOException;

final class Continuation extends PrologObject {
	Operation code;
	Term Args[];

	Continuation(Term args[], Code c) {
		int i = c.Arity() + 1;
		Args = new Term[i];
		while (i-- > 0)
			Args[i] = args[i];
		code = c;
	}

	public Continuation(Term[] args, int getArity, Operation c) {
		int i = getArity + 1;
		Args = new Term[i];
		while (i-- > 0)
			Args[i] = args[i];
		code = (Code) c;
	}

	public Term Deref() {
		return this;
	}

	public boolean Unify(Term that, Prolog mach) {
		return that.Bind(this, mach);
	}

	public void formattedOutput(int formatFlags, Appendable buffer) throws IOException {
		System.out.println("general output of continuations not available");
		buffer.append(GetName());
		int i = Args.length;
		int current = 0;
		while (i-- > 0) {
			formattedArgOutput(Args[current++], formatFlags, buffer);
		}

	}

	private void formattedArgOutput(Term term, int formatFlags, Appendable buffer) throws IOException {
		if (term != null)
			term.formattedOutput(formatFlags, buffer);

	}

	@Override
	public int Arity() {
		return JAVA;
	}

}